### ISSUE TITLE
Deprecate `mbedtls_cipher_setup_psa()`

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#5261]
After #5204 there is no longer any part of the library that calls this function. Since it was mainly created for internal use, let's deprecate it and encourage users to actually migrate to the PSA API instead (see #5208 too).

- [x] Mark the function as deprecated using `MBEDTLS_DEPRECATED` (see `platform_util.h`)
- [x] Document the deprecation using `\deprecated`
- [x] Add a `New deprecations` ChangeLog entry that recommends use `psa_aead_xxx()` / `psa_cipher_xxx()` directly instead.

Note: updating `docs/use-psa-crypto.md` is out of scope, as it's already done in #5757
